src/ASM/ASMu2DLag.C

```diff
@@ -177,6 +177,44 @@ int ASMu2DLag::parseNodeBox (const std::string& setName, const char* data)
 }
 
 
+int ASMu2DLag::getElementSetIdx (const std::string& setName) const
+{
+  int idx = 1;
+  for (const ASM::NodeSet& es : elemSets)
+    if (es.first == setName)
+      return idx;
+    else
+      ++idx;
+
+  return 0;
+}
+
+
+const IntVec& ASMu2DLag::getElementSet (int idx) const
+{
+  int count = 0;
+  for (const ASM::NodeSet& es : elemSets)
+    if (++count == idx)
+      return es.second;
+
+  return this->ASMbase::getElementSet(idx);
+}
+
+
+IntVec& ASMu2DLag::getElementSet (const std::string& setName, int& idx)
+{
+  idx = 1;
+  for (ASM::NodeSet& es : nodeSets)
+    if (es.first == setName)
+      return es.second;
+    else if (idx)
+      ++idx;
+
+  elemSets.push_back(std::make_pair(setName,IntVec()));
```

Comment:
```c++
elemSets.emplace_back(setName,IntVec())
```

Reply:
done (still not got used to using emplace_back() instead of push_back()).